scripts/ghutils.py

```diff
@@ -92,5 +111,219 @@ def add_issue_url_to_pr(repo, pull_request_number, issue_key):
             raise Exception("There was a failure updating the pull request.")
 
 
+def branch_api_request(repo, branch, method, postfix="", data=None):
+    """Make Requests to the github branch protection api.
+
+    Not supported by github3.py yet (6th September 2017)
+    """
+    url = "{branch_url}/protection{postfix}".format(
+        branch_url=repo.branches_urlt.expand(branch=branch),
+        postfix=postfix
+    )
+    # Branch protection api is in preview and requires a specific content type
+    response = repo._session.request(
+        method, url,
+        headers={'Accept': 'application/vnd.github.loki-preview+json'},
+        data=data)
+    return response
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    '--mainline',
+    required=True,
+    help="Mainline branch to cut from"
+)
+@click.option(
+    '--rc',
+    help="Release Candidate branch (re)create"
+)
+def update_rc_branch(ctx, mainline, rc):
+    """Update rc branch.
+
+    1. Store branch protection data
+    2. Delete rc branch
+    3. Create rc branch from head of mainline
+    4. Enable branch protection with skeleton or previously stored settings.
+    """
+    repo = ctx.obj
+    rc = try_context(repo, rc, "rc", "rc_ref")
+    branch_protection_enabled = False
+
+    # check if branch exists
+    if rc in (b.name for b in repo.iter_branches()):
+        logger.debug("Branch {} exists".format(rc))
+        # rc branch exists
+        branch_protection_response = branch_api_request(repo, rc, 'GET')
+        if branch_protection_response.status_code == 200:
+            # rc branch exists and protection enabled
+            logger.debug("Branch {} has protection enabled".format(rc))
+            branch_protection_enabled = True
+            # disable branch protection
+            r = branch_api_request(repo, rc, 'DELETE')
+            r.raise_for_status()
+            logger.debug("Branch protection disabled")
+        elif branch_protection_response.status_code == 404:
```

Comment:
Why would branch protection not be enabled? I don't think this should be ignored, it should be required.

Reply:
It is required, but at this point we are trying to disable it, if its already disabled, nothing to do. 
Once the push has happened, branch protection will be enabled regardless of whether it was enabled previously.

Reply:
Yeah, I missed that.